jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -0,0 +1,258 @@
+    {
+        ServletHolder defaultHolder = context.addServlet(DefaultServlet.class, "/static/*");
+        defaultHolder.setInitParameter("pathInfoOnly", String.valueOf(pathInfoOnly));
+        defaultHolder.setInitParameter("dirAllowed", "false");
+
+        server.setHandler(context);
+        server.addConnector(connector);
+
+        server.start();
+        ((DefaultServlet)defaultHolder.getServlet()).getResourceService().setWelcomeMode(welcomeMode);
+    }
+
+    @AfterEach
+    public void destroy() throws Exception
+    {
+        server.stop();
+        server.join();
+    }
+
+    public static class WelcomeServlet extends HttpServlet
+    {
+        @Override
+        protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+        {
+            resp.setStatus(200);
+            String mapping = getInitParameter("mapping");
+            resp.getWriter().print("Servlet at " + mapping);
+        }
+    }
+
+    record Data(boolean pathInfoOnly, ResourceService.WelcomeMode welcomeMode, String requestPath, int expectedStatus, String expected)
+    {
+    }
+
+    public static Stream<Data> data()
+    {
+        List<Data> datas = new ArrayList<>();
+        for (String requestPath : List.of("/foo.welcome", "/static/foo.welcome", "/static/", "/static/subdirHtml/",
+            "/static/subdirWelcome/", "/static/subdirEmpty/", "/empty/", "/nothing/index.welcome", "/nothing/"))
+        {
+            for (ResourceService.WelcomeMode welcomeMode : List.of(SERVE, REDIRECT, REHANDLE))
+            {
+                for (boolean pathInfoOnly : List.of(false, true))
+                {
+                    int expectedStatus;
+                    String expected;
+
+                    switch (requestPath)
+                    {
+                        case "/foo.welcome" ->
+                        {
+                            expectedStatus = HttpStatus.OK_200;
+                            expected = "Servlet at welcome extension";
+                        }
+                        case "/static/foo.welcome" ->
+                        {
+                            expectedStatus = HttpStatus.OK_200;
+                            expected = pathInfoOnly ? "Static foo.welcome at root" : "Static foo.welcome at static";
+                        }
+                        case "/static/" ->
+                        {
+                            if (welcomeMode == REDIRECT)
+                            {
+                                expectedStatus = HttpStatus.FOUND_302;
+                                expected = pathInfoOnly ? "http://local/static/static/index.html" : "http://local/static/index.html";
+                            }
+                            else
+                            {
+                                expectedStatus = HttpStatus.OK_200;
+                                expected = pathInfoOnly ? "Static index.html at root" : "Static index.html at static";
```

Comment:
```suggestion
                                expected = "Static index.html at static";
```

Reply:
In my understanding, when `pathInfoOnly == true`, the servlet path (`/static`) is supposed to be stripped from the path, so `docroot/index.html` should be served instead of `docroot/static/index.html`.

Or did I misunderstand?

Reply:
No - I did!